# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1668159712 -3600
# Parent  eed827f2d3fcb92f59128f3e4294b4657ce88821
Bug 1800202 - Close db after use in nsMsgDBFolder::ReadDBFolderInfo().

This is run from discoverFolders() at startup and can cause MSF loss if too many folders are left open.

diff --git a/mailnews/base/src/nsMsgDBFolder.cpp b/mailnews/base/src/nsMsgDBFolder.cpp
--- a/mailnews/base/src/nsMsgDBFolder.cpp
+++ b/mailnews/base/src/nsMsgDBFolder.cpp
@@ -637,16 +637,17 @@ nsresult nsMsgDBFolder::ReadDBFolderInfo
         }
       }
     }
   }
 
   if (force || !mInitializedFromCache) {
     nsCOMPtr<nsIDBFolderInfo> folderInfo;
     nsCOMPtr<nsIMsgDatabase> db;
+    bool weOpenedDB = !mDatabase;
     result =
         GetDBFolderInfoAndDB(getter_AddRefs(folderInfo), getter_AddRefs(db));
     if (NS_SUCCEEDED(result)) {
       if (folderInfo) {
         if (!mInitializedFromCache) {
           folderInfo->GetFlags((int32_t*)&mFlags);
 #ifdef DEBUG_bienvenu1
           nsString name;
@@ -671,16 +672,17 @@ nsresult nsMsgDBFolder::ReadDBFolderInfo
         // folderInfo->GetImapUnreadPendingMessages(&mNumPendingUnreadMessages);
 
         if (db) {
           bool hasnew;
           nsresult rv;
           rv = db->HasNew(&hasnew);
           if (NS_FAILED(rv)) return rv;
         }
+        if (weOpenedDB) CloseDBIfFolderNotOpen(false);
       }
     } else {
       // we tried to open DB but failed - don't keep trying.
       // If a DB is created, we will call this method with force == TRUE,
       // and read from the db that way.
       mInitializedFromCache = true;
     }
   }
